Create How-To-Jellyfin-Template-plugin.txt (#26)
jellyfin/jellyfin-plugin-template: DEVELOPMENT.md

# How-To Compile, Install & Run Jellyfin Template plugin
  
  
document creation date: 01/01/2021
  
## Intro
  

> 1. Jellyfin is a Plex OpenSource equivalent, to read more about it goto: https://jellyfin.org  

Reviewer: ```suggestion
1. Jellyfin is an open source media server, to read more about it, go to: https://jellyfin.org  
```

Don't think we really need the plex comparison if someone is attempting to make a jellyfin plugin

# How-To Compile, Install & Run Jellyfin Template plugin
  
  
document creation date: 01/01/2021
  
## Intro
  
1. Jellyfin is an open source media server, to read more about it, go to: https://jellyfin.org  

> 2. Jellyfin has plugin infrastructure to expand and enhance it's capabilities, plugin are written in C# and Html  

Reviewer: ```suggestion
2. Jellyfin has a plugin infrastructure to expand and enhance its capabilities. Plugins are written in C# and Html.
```

> 3. A good staring point for Jellyfin Plugin Development is the 'Template' project found at: https://github.com/jellyfin/jellyfin-plugin-template  

Reviewer: ```suggestion
3. A good starting point for Jellyfin Plugin Development is the 'Template' project found at: https://github.com/jellyfin/jellyfin-plugin-template  
```

> 4. the template project needs more info for beginners, this readme file is a step by step guide to build and install 'Template' plugin project  

Reviewer: ```suggestion
4. The template project needs more info for beginners. This readme file is a step by step guide to build and install 'Template' plugin project
```

15. make sure you see plugins tiles there that are ready to be installed (and can't see your 'Template' plugin yet)  
16. make sure the plugins folder exist (in windows: `%UserProfile%\AppData\Local\jellyfin\plugins`)  
  
  
## Download the 'Template' plugin solution
  
1. goto: https://github.com/jellyfin/jellyfin-plugin-template  
2. from the top bar click the 'Code' button and press the 'Download ZIP' sub menu button  
3. once the zip has downloaded - extract it to a folder (lets say on windows: `%UserProfile%\Downloads\pluginTemplate`)  
4. in that folder there should be a folder called: jellyfin-plugin-template-master and in it you should find:  
	- Jellyfin.Plugin.Template
	- .editorconfig
	- .gitignore
	- build.yaml
	- Jellyfin.Plugin.Template.sln
	- LICENSE
	- README.md
	- ...  
5. move the extracted folder (jellyfin-plugin-template-master) to the folder you want the solution to run from (for example: `C:\workspace\jellyfin-plugin-template-master`), we will refer to this folder as: `[TEMPLATE_PLUGIN_PROJECT_ROOT_FOLDER]`  
  
  
## Open the 'Template' plugin solution in latest 'Microsoft Visual Studio'
  
>
> [!NOTE] 

> > For this section you can either use: '**Microsoft Visual Studio**',

Reviewer 2: Visual Studio Code is not the free community version of Visual Studio. They are 2 completely different programs.
[Visual Studio Community Edition](https://visualstudio.microsoft.com/vs/community/) is the free community version of Visual Studio.

## Stages 10 - 15 are required for 17.0RC1 & 17.0RC2 ONLY
10. stop the server  
11. edit the 'meta.json' file with a text editor
12. find the line: "guid": "...",
13. replace the GUID string with your created GUID  
14. save the file  
15. start the server again  
  
## Check your Template Plugin installed
  
1. goto: http://localhost:8096  
2. login to the server using the recently created user (that you created in the wizard), sometimes it will require to restart the Jellyfin.exe app  
3. once logged in, click the 3-line menu button from the top left corner (sometimes reffered to as 'Hamburger')  
4. Click the 'Dashboard' submenu button  
5. scroll down and click the 'Plugins' submenu button  
6. make sure you see your 'Template' plugin as a tile and it has "status:active" underneath the names.
7. if any problems occurr, if it says "status:incompatible", or you can't see your Template plugin then goto: `%UserProfile%\AppData\Local\jellyfin\log` and open the latest log file and try to search for the plugin file name and see if there is an error or exception (then google it)  
  
after this goto: https://github.com/jellyfin/jellyfin-plugin-template

> to learn more about JellyFin Plugins and how to implement content and functionality into them,

Reviewer: ```suggestion
To learn more about Jellyfin Plugins and how to implement content and functionality for them.
```
